android/src/ti/bluetooth/TiBluetoothModule.java

```diff
@@ -8,301 +8,254 @@
-        }
-    };
-
-
-    private List<ScanFilter> scanFilters(String[] ids) {
-        List<ScanFilter> list = new ArrayList<ScanFilter>(1);
-
-        for (int i = 0; i < ids.length; i++) {
-            ScanFilter filter = new ScanFilter.Builder().setServiceUuid(ParcelUuid.fromString(ids[i])).build();
-            list.add(filter);
-        }
-        return list;
-    }
-
-    @Kroll.getProperty
-    @Kroll.method
-    public int getState() {
-        return currentState;
-    }
-
-    @Kroll.getProperty
-    @Kroll.method
-    public boolean isScanning() {
-        return isScanning;
-    }
-
-    @Kroll.method
-    public void initialize() {
-        // TODO check if permissions are correctly set:
-        //int permissionCheck = activity.checkSelfPermission(Manifest.permission.ACCESS_COARSE_LOCATION);
-        btManager = (BluetoothManager)appContext.getSystemService(appContext.BLUETOOTH_SERVICE);
-        btAdapter = btManager.getAdapter();
-        if (btAdapter != null) {
-            Log.d(LCAT, "BT init");
-            currentState = btAdapter.getState();
-        } else {
-            currentState = MANAGER_STATE_UNSUPPORTED;
-        }
-    }
-
-    @Kroll.getProperty
-    @Kroll.method
-    public int getScanMode() {
-        return currentScanMode;
-    }
-
-    @Kroll.setProperty
-    @Kroll.method
-    public void setScanMode(int scanMode) {
-        if (scanMode == SCAN_MODE_BALANCED || scanMode == SCAN_MODE_LOW_POWER || scanMode == SCAN_MODE_LOW_LATENCY || scanMode == SCAN_MODE_OPPORTUNISTIC) {
-            currentScanMode = scanMode;
-        } else {
-            currentScanMode = SCAN_MODE_LOW_POWER;
-        }
-
-    }
-
-    @Kroll.method
-    public void startScan() {
-        // start scanning for every device
-        if (btAdapter != null) {
-            ScanSettings settings = new ScanSettings.Builder().setScanMode(currentScanMode).build();
-            btScanner = btAdapter.getBluetoothLeScanner();
-            btScanner.startScan(new ArrayList<ScanFilter>(), settings, scanCallback);
-            isScanning = true;
-            Log.i(LCAT,"Start scan");
-        }
-    }
-
-    @Kroll.method
-    public void startScanWithServices(String[] obj) {
-        // start scanning for a list of devices
-        if (btAdapter != null) {
-            ScanSettings settings = new ScanSettings.Builder().setScanMode(currentScanMode).build();
-            btScanner = btAdapter.getBluetoothLeScanner();
-            btScanner.startScan(scanFilters(obj), settings, scanCallback);
-            isScanning = true;
-        }
-    }
-
-    @Kroll.method
-    public void stopScan() {
-        // stop scanning
-        if (btAdapter != null) {
-            btScanner.stopScan(scanCallback);
-            isScanning = false;
-        }
-    }
+	public interface ConnectionCallback {
+		void onConnectionStateChange(BluetoothDevice device, int newState);
+	}
+
+	public static final String LCAT = "BLE🔵";
```

Comment:
Please remove the blue dot:
![ble](https://cloud.githubusercontent.com/assets/4334997/21577468/c34a0ff2-cf5b-11e6-9198-ee06786aac3c.png)